Home Connect: migrate actions to dedicated action pages (#46046)
home-assistant/home-assistant.github.io: source/_actions/home_connect.set_program_and_options.markdown

---
title: "Set program and options"
action: home_connect.set_program_and_options
domain: home_connect
description: "Starts or selects a program on a Home Connect appliance, or sets the options for the active or selected program."
related_actions:
  - home_connect.start_selected_program
  - home_connect.change_setting
---

> Use this action to start or select a program on a Home Connect appliance, for example to start an eco dishwasher cycle or pick a coffee program ready to brew. If you leave the program out, the action instead sets options on the program that is already active or selected, so you can tweak a running cycle.

Reviewer: Let's use a clearer word:

```suggestion
Use this action to start or select a program on a Home Connect appliance, for example to start an eco dishwasher cycle or pick a coffee program ready to brew. If you omit the program, the action instead sets options on the program that is already active or selected, so you can tweak a running cycle.
```

3. If you're setting up a new automation, add a trigger in the **When** section. Scripts don't need a trigger. They run when something else calls them.
4. In the **Then do** section, select **Add action**.
5. From the list of actions, search for and select **Set program and options**.
6. Select the appliance you want to control.
7. Under **Affects to**, choose whether the action affects the active or the selected program.
8. Optionally, choose a program and set any options for your appliance.
9. Select **Save**.

### Options in the UI

{% options_ui %}
Device:
  description: The Home Connect appliance you want to control.
Affects to:
  description: Whether the action affects the active program (the one currently running) or the selected program (the one queued to run). Choose **Active program** or **Selected program**.
Program:
  description: The program to start or select. When set, the appliance starts or selects this program depending on the **Affects to** value. When left empty, the action only sets options on the active or selected program.
  required: false
{% endoptions_ui %}

> In addition, the UI shows option groups for each appliance type. Only the options that apply to your appliance are used. See [Program options](#program-options) for the full list.

Reviewer: Start sentence with the goal, then action:

```suggestion
In addition, the UI shows option groups for each appliance type. Only the options that apply to your appliance are used.
For the full list, see [Program options](#program-options).
```

{% include actions/yaml_header.md %}

In YAML, refer to this action as `home_connect.set_program_and_options`. A basic example that starts an eco dishwasher program looks like this:

{% example %}
action: |
  action: home_connect.set_program_and_options
  data:
    device_id: a1b2c3d4e5f6a7b8c9d0e1f2a3b4c5d6
    affects_to: active_program
    program: dishcare_dishwasher_program_eco_50
{% endexample %}

### Options in YAML

{% options_yaml %}
device_id:
  description: The device ID of the Home Connect appliance you want to control.
  required: true
  type: string
affects_to:
  description: Whether the action affects the active program or the selected program. Use `active_program` or `selected_program`.
  required: true
  type: string
program:
  description: The program to start or select. When set, the appliance starts or selects this program depending on the `affects_to` value. When left out, the action only sets options on the active or selected program.
  required: false
  type: string
{% endoptions_yaml %}

> In addition, you can pass program options as extra keys in the `data` section. Only the options that apply to your appliance are used. See [Program options](#program-options) for the full list.

Reviewer: ```suggestion
In addition, you can pass program options as extra keys in the `data` section. Only the options that apply to your appliance are used.
For the full list, see [Program options](#program-options).
```
